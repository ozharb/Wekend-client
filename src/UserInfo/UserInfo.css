
.row1{
    display: flex;

}
.list{
    font-family: 'Public Sans', sans-serif;

  
}
.menu-items{
    display: flex;
    justify-content: flex-start;

}

.dayboard{
    text-decoration: none;
    font-family: 'Public Sans', sans-serif;
    writing-mode: vertical-rl;
    text-orientation: upright;
    border: solid ;
    padding-top: 10px;
  margin-top: 37px;
 margin-bottom: 18px;
  background: orange; 

  margin-right: 3px;
}
li>a{
    text-decoration: none;
}
.list>.side2>li{
   
    padding: 5px;
    border-style: solid;
    text-align: center;
    margin: 2px;
}
.row1>li{
    padding: 5px;
    border-style: solid;
    text-align: center;
    margin: 2px;
}
.li-add-event{
    background: #fbff12;
}
.li-filter{
    background: #ff206e;
}
.li-friends{
    background: #41ead4;
}
.li-logout{
    background: #9EF573;
}
h4{
    margin: 0;
}
.logout{
    font-size: 15px;
}
