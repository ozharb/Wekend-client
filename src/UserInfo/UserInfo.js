import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import './UserInfo.css'

import TokenService from '../services/token-service'


class UserInfo extends Component {
  

handleLogoutClick = () => {
    let username = 'user'
    TokenService.clearAuthToken()
    window.localStorage.removeItem(username)

  }

render(){

    let username = 'user'
    const user =  window.localStorage.getItem(username)
  return (
    <div className = 'menu-items'>

  <Link className='dayboard' to='/days'>home</Link>    


    <ul className="list">
   
Hi {user}


<div className='side2'>
  <li className="li-add-event"><Link to = '/new-event'>Add Event</Link></li>
<div className='row1'>
  <li className="li-filter"><Link to = '/filter'>Filter</Link></li>
  <li className="li-friends"><Link to='/friends'>Friends</Link></li>
</div>
<li className="li-logout"><Link
onClick={this.handleLogoutClick}
to='/login'
className = "logout">
Logout
</Link> </li>
</div>
   </ul>
 </div>

  )
}
}

export default UserInfo