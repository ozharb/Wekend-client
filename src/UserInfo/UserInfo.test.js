import React from 'react';
import UserInfo from './UserInfo';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

it('renders the UI as expected', () => {
  const tree = renderer
  
    .create(
      <BrowserRouter>  
    <UserInfo  />
    </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });