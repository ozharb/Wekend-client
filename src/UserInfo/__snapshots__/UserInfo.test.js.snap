// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="menu-items"
>
  <a
    className="dayboard"
    href="/days"
    onClick={[Function]}
  >
    home
  </a>
  <ul
    className="list"
  >
    Hi 
    <div
      className="side2"
    >
      <li
        className="li-add-event"
      >
        <a
          href="/new-event"
          onClick={[Function]}
        >
          Add Event
        </a>
      </li>
      <div
        className="row1"
      >
        <li
          className="li-filter"
        >
          <a
            href="/filter"
            onClick={[Function]}
          >
            Filter
          </a>
        </li>
        <li
          className="li-friends"
        >
          <a
            href="/friends"
            onClick={[Function]}
          >
            Friends
          </a>
        </li>
      </div>
      <li
        className="li-logout"
      >
        <a
          className="logout"
          href="/login"
          onClick={[Function]}
        >
          Logout
        </a>
         
      </li>
    </div>
  </ul>
</div>
`;
